File: src/main/java/com/monkeyteam/monkeyutils/modules/CropsBreak.java
package com.monkeyteam.monkeyutils.modules;

import org.bukkit.Material;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.block.Action;
import org.bukkit.event.player.PlayerInteractEvent;

public class CropsBreak implements Listener {

    @EventHandler
    public void onBreakCrop(PlayerInteractEvent e) {
        if (e.getAction() == Action.PHYSICAL) {
            try {
                if (e.getClickedBlock().getType() == Material.FARMLAND) {
                    e.setCancelled(true);
                }
            } catch (NullPointerException ignored) {
            }
        }
    }
}

File: src/main/java/com/monkeyteam/monkeyutils/modules/CropsBreak.kt
package com.monkeyteam.monkeyutils.modules

import org.bukkit.Material
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.block.Action
import org.bukkit.event.player.PlayerInteractEvent

object CropsBreak: Listener {
    @EventHandler
    fun onBreakCrop(e: PlayerInteractEvent) {
        if (e.action == Action.PHYSICAL) {
            try {
                if (e.clickedBlock!!.type == Material.FARMLAND) {
                    e.isCancelled = true
                }
            } catch (ignored: NullPointerException) {
            }
        }
    }
}